src/overlays/actors/ovl_Elf_Msg/z_elf_msg.c

```diff
@@ -21,23 +54,234 @@ const ActorInit Elf_Msg_InitVars = {
+    }
+
+    if (!ElfMsg_KillCheck(this, globalCtx)) {
+        Actor_ProcessInitChain(thisx, sInitChain);
+        if (thisx->posRot.rot.x == 0) {
+            thisx->scale.z = 0.4f;
+            thisx->scale.x = 0.4f;
+        } else {
+            thisx->scale.x = thisx->scale.z = thisx->posRot.rot.x * 0.04f;
+        }
+
+        if (thisx->posRot.rot.z == 0) {
+            thisx->scale.y = 0.4f;
+        } else {
+            thisx->scale.y = thisx->posRot.rot.z * 0.04f;
+        }
+
+        if (thisx->params & 0x4000) {
+            ElfMsg_SetupAction(this, ElfMsg_CallNaviCuboid);
+        } else {
+            ElfMsg_SetupAction(this, ElfMsg_CallNaviCylinder);
+        }
+
+        thisx->shape.rot.x = thisx->shape.rot.y = thisx->shape.rot.z = 0;
+    }
+}
+
+void ElfMsg_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+s32 ElfMsg_GetMessageId(ElfMsg* this) {
+    // Negative message ID forces link to talk to Navi
+    if (this->actor.params & 0x8000) {
+        return (this->actor.params & 0xFF) + 0x100;
+    } else {
+        return -0x100 - (this->actor.params & 0xFF);
+    }
+}
+
+void ElfMsg_CallNaviCuboid(ElfMsg* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    EnElf* navi = (EnElf*)player->navi;
+
+    if ((fabsf(player->actor.posRot.pos.x - this->actor.posRot.pos.x) < (100.0f * this->actor.scale.x)) &&
+        (this->actor.posRot.pos.y <= player->actor.posRot.pos.y) &&
+        ((player->actor.posRot.pos.y - this->actor.posRot.pos.y) < (100.0f * this->actor.scale.y)) &&
+        (fabsf(player->actor.posRot.pos.z - this->actor.posRot.pos.z) < (100.0f * this->actor.scale.z))) {
+        player->naviMessageId = ElfMsg_GetMessageId(this);
+        navi->unk_298 = &this->actor;
+    }
+}
+
+s32 ElfMsg_IsInNaviRangeXZ(Vec3f* playerPos, Vec3f* msgPos, f32 radius) {
+    return (SQ(msgPos->x - playerPos->x) + SQ(msgPos->z - playerPos->z)) < SQ(radius);
+}
+
+void ElfMsg_CallNaviCylinder(ElfMsg* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    EnElf* navi = (EnElf*)player->navi;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Elf_Msg/ElfMsg_Init.s")
+    if (ElfMsg_IsInNaviRangeXZ(&player->actor.posRot.pos, &this->actor.posRot.pos, this->actor.scale.x * 100.0f) &&
+        (this->actor.posRot.pos.y <= player->actor.posRot.pos.y) &&
+        ((player->actor.posRot.pos.y - this->actor.posRot.pos.y) < (100.0f * this->actor.scale.y))) {
+        player->naviMessageId = ElfMsg_GetMessageId(this);
+        navi->unk_298 = &this->actor;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Elf_Msg/ElfMsg_Destroy.s")
+void ElfMsg_Update(Actor* thisx, GlobalContext* globalCtx) {
+    ElfMsg* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Elf_Msg/func_809ACDCC.s")
+    if (!ElfMsg_KillCheck(this, globalCtx)) {
+        // if actor flag 0x100 is set:
+        if (func_8002F194(&this->actor, globalCtx)) {
```

Comment:
Same as my prevous comment, Should I remove this now becuase the function will have a self-explanatory name later?

Reply:
im impartial to these. they provide good context while the functions are unnamed, but it creates more work to remove them when the function is named proper

Reply:
I think comments like that should be omitted, these very commonly used functions will be named before we get to a point where specific actor documentation really matters.